xbrowse_server/phenotips/views.py

```diff
@@ -1,90 +1,187 @@
+        auth_level = 'editor'
+    elif project.is_public:
+        auth_level = 'public'
+    elif project.can_view(request.user):
+        auth_level = 'viewer'
+    else:
+        raise Exception("Authx - cannot authenticate:fetch_phenotips_edit_page ")
+    request.session['current_auth_level']=auth_level
+    
+  else: 
+    #getting admin account to translate eid to id
+    project_name = request.session['current_project_name']
+    ext_id=request.session['current_ext_id']
+    auth_level = request.session['current_auth_level']
+    
+  #depending on auth level, pick either the full edit username or the view-only username for this
+  #project to fetch page
+  phenotips_uname,phenotips_pwd ='_','_'
+  if auth_level=='admin':
+    phenotips_uname,phenotips_pwd = get_uname_pwd_for_project(project_name)
+  else:
+    phenotips_uname,phenotips_pwd = get_generic_collaborator_uname_pwd_for_project(project_name)
+  url= settings.PHENOPTIPS_HOST_NAME+'/bin/'+ ext_id
+  if auth_level=='admin':
+    url= settings.PHENOPTIPS_HOST_NAME+'/bin/edit/data/'+ ext_id
+  if not request.GET.has_key('project'):
+    url += '?'
+    counter=0
+    for param,val in request.GET.iteritems():
+      url += param + '=' + val
+      if counter < len(request.GET)-1:
+        url += '&'
+      counter+=1
+  if type(ext_id) != dict:
+    #we are using the project name as the username and project name twice as the password
+    #for example if project name was foo, the username would be foo, password would be foofoo
+    result = do_authenticated_call_to_phenotips(url,phenotips_uname,phenotips_pwd)
+    response = __add_back_phenotips_headers_response(result)
+    return response
+  else:
+    logger.error('phenotips.views:'+ext_id['error'])
+    raise Http404    
 
 
+#do a GET as a proxy for Phenotips
+#exempting csrf here since phenotips doesn't have this support
+@csrf_exempt
+def proxy_get(request):
+  '''to act as a proxy for get requests '''
+  project_name = request.session['current_project_name']
+  project_phenotips_uname,project_phenotips_pwd = get_uname_pwd_for_project(project_name)
+  try:
+    result = do_authenticated_call_to_phenotips(__aggregate_url_parameters(request),project_phenotips_uname,project_phenotips_pwd)
+    return __add_back_phenotips_headers_response(result)
+  except Exception as e:
+    print 'proxy get error:',e
+    logger.error('phenotips.views:'+str(e))
+    raise Http404
+
+
+#given a request object,and base URL aggregates and returns a reconstructed URL
+def __aggregate_url_parameters(request):
+  '''given a request object, aggregates and returns parameters'''
+  counter=0
+  url=settings.PHENOPTIPS_HOST_NAME+request.path + '?'
+  for param,val in request.GET.iteritems():
+    url += param + '=' + val
+    if counter < len(request.GET)-1:
+      url += '&'
+    counter+=1
+  return url
+
+#do a POST as a proxy for Phenotips
+#exempting csrf here since phenotips doesn't have this support
+@csrf_exempt
```

Comment:
Done and committed. Didn't think it needed one here since it has no URL mappings. but good to be safe. Added logging too.